aggregator/src/aggregator/collect_job_driver.rs

```diff
@@ -459,17 +459,24 @@ impl CollectJobDriver {
         lease_duration: Duration,
     ) -> impl Fn(usize) -> BoxFuture<'static, Result<Vec<Lease<AcquiredCollectJob>>, datastore::Error>>
     {
-        move |maximum_acquire_count| {
+        move |maximum_acquire_count_per_query_type| {
             let datastore = Arc::clone(&datastore);
             Box::pin(async move {
                 datastore
                     .run_tx(|tx| {
                         Box::pin(async move {
-                            tx.acquire_incomplete_collect_jobs(
-                                &lease_duration,
-                                maximum_acquire_count,
-                            )
-                            .await
+                            let (mut time_interval_jobs, mut fixed_size_jobs) = try_join!(
+                                tx.acquire_incomplete_time_interval_collect_jobs(
+                                    &lease_duration,
+                                    maximum_acquire_count_per_query_type,
+                                ),
+                                tx.acquire_incomplete_fixed_size_collect_jobs(
+                                    &lease_duration,
+                                    maximum_acquire_count_per_query_type
+                                ),
+                            )?;
+                            time_interval_jobs.append(&mut fixed_size_jobs);
```

Comment:
nit: using [`Vec::extend`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extend) would not require `fixed_size_jobs` to be `mut`.

Reply:
SG, I went with an approach that chained two iterators to avoid making either variable `mut`.